### ISSUE TITLE
tsqllint not compliant with dotnet 6.0

### ISSUE THREAD

**Issue by nvuillam** [tsqllint/tsqllint#273]
Is it planned to make tsqllint compliant with dotnet 6.0.0 ?

As tsqllint is integrated within MegaLinter, I have a compatibility issue with latest version of dotnet that I installed the following way on an alpine linux image

```dockerfile
RUN wget --tries=5 -q -O dotnet-install.sh https://dot.net/v1/dotnet-install.sh \
          && chmod +x dotnet-install.sh \
          && ./dotnet-install.sh --install-dir /usr/share/dotnet -channel Current -version latest
ENV PATH="${PATH}:/root/.dotnet/tools:/usr/share/dotnet"
RUN dotnet tool install --global TSQLLint
```

The result of call to tsqllint is the following

```shell
[tsqllint] command: ['tsqllint', '/tmp/lint/.automation/test/sql_tsql/sql_good_1.sql']
[tsqllint] CWD: /
[tsqllint] result: 150 It was not possible to find any compatible framework version
The framework 'Microsoft.NETCore.App', version '5.0.0' (x64) was not found.
  - The following frameworks were found:
      6.0.0 at [/usr/share/dotnet/shared/Microsoft.NETCore.App]

You can resolve the problem by installing the specified framework and/or SDK.
```

**Comment by echoix** [comment]
I looked at this Friday. To start with, I upgraded the tsqllint_common dependency, by essentially using the upgrade-assistant and adding a net6.0 target framework (dotnet 6 can compile and target for net5.0 too (netcoreapp5.0 was the old name). This makes the whole project/solution modernized. And then the same thing with this project, but had to make a little more changes. The one dependency that trails behind and thus needs to package some more older netframework 4.* files is te CommandLineParser, that seems abandoned for 1.5-2 years, whereas it has a big community wanting to continue fixing it. Its sad, it seemed to have been common usage for years and years. I was not able to fully understand how to migrate from the 1.9.x versions (from 2013) to the 2.x versions from up to late 2019, since I didn't understand the migrating documents to fix the breaking changes. But it still works. There are many more warnings though, since the C# language evolved a little bit.
For now, I'm just trying to get the tests running on the current state on master, to compare what is different, since on master they skip all 330 tests...
